src/main/java/org/pac4j/vertx/handler/impl/RequiresAuthenticationHandler.java

```diff
@@ -138,6 +145,33 @@ public void handle(RoutingContext routingContext) {
 
     }
 
+    @Override
+    protected void authorise(final User user, final RoutingContext context) {
+        if (! (user instanceof Pac4jUser)) {
+            throw new TechnicalException("Wrong user type in authorise");
+        }
+
+        Handler<AsyncResult<Boolean>> authHandler = res -> {
+            if (res.succeeded()) {
+                if (res.result()) {
+                    context.next();
+                } else {
+                    context.fail(403);
+                }
+            } else {
+                context.fail(res.cause());
```

Comment:
One more thing: maybe it's already handled somewhere, but I don't see the httpAdapter to handle the 403 and 500 errors, like I have in Play.

-> https://github.com/pac4j/play-pac4j/blob/master/play-pac4j-java/src/main/java/org/pac4j/play/java/RequiresAuthenticationAction.java#L177